src/Data/Singletons/Promote.hs

```diff
@@ -387,10 +444,11 @@ promoteInstanceDec orig_meth_sigs cls_tvbs_map
                                   , id_sigs     = inst_sigs
                                   , id_meths    = meths }) = do
   opts <- getOptions
-  cls_tvb_names <- lookup_cls_tvb_names
+  cls_tvbs <- lookup_cls_tvbs
```

Comment:
I originally thought that I would need to use the class-bound `TyVarBndr`s in the SAKS for promoted instance methods, so these changes were made in pursuit of that goal. However, I later realized that the SAKS for promoted instance methods don't need to explicitly bind kind variables at all—see my comment starting with "`For class method RHS helper functions, don't bother quantifying`..." in `promote_let_dec_ty`. As a result, this `TyVarBndr` refactoring went unused.

I still kind of like the new code over the old code, since we now only invoke `map extractTvbName` once instead of twice. But if you'd prefer I can revert this change.

Reply:
You can leave it changed. It was more that I was worried I was missing something critical.